### DISCUSSION TITLE
5.11.1 bump version

### DISCUSSION THREAD

**Discussion by dnyandeobharambe** [GitTools/GitVersion#3986] [Q&A] [answered]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

GitVersion.Tool

### GitVersion version

5.11.1

### Operating system

Linux

### What are you seeing?

Can not increase version when I have comment as \+semver:patch  here is my config file
mode: Mainline
major-version-bump-message: '\+semver:\s?(breaking|major)'
minor-version-bump-message: '\+semver:\s?(feature|minor)'
patch-version-bump-message: '\+semver:\s?(fix|patch)'
no-bump-message: '\+semver:\s?(none|skip)'
commit-message-incrementing: Enabled
branches:
  release:
    regex: ^releases?[/-]
    tag: ''
  feature:
    regex: ^feature?[/-]
    tag: ''

### What is expected?

Initially branch is tag with 0.0.1 now it expect to go as 0.0.2
But somehow it take  8162023.0.11 and it keep getting same everytime. 
I delete all tags  from repository but not sure from where it take.
Not sure how to fix it

### Steps to Reproduce

Configure as above and then run the pipeline , it support to increase it. But looks like this part never increase it
and this is part of your code, I am just calling it

- task: gitversion/setup@0
      displayName: 'Install GitTools'
      inputs:
        versionSpec: '5.11.1'
    - task: gitversion/execute@0
      name: CalculateSemVer
      displayName: 'Calculate SemVer'
      inputs:
        useConfigFile: true
        configFilePath: './azure-pipelines/GitVersion.yml'
    - bash: |
        currentVersion=$(GitVersion.SemVer)
        echo "currentversion is $currentVersion"
        echo "##vso[task.setvariable variable=semversion;isOutput=true;]$currentVersion"

### RepositoryFixture Test

NA

### Output log or link to your CI build (if appropriate).

```shell
NA
```

**Comment by BinToss** [comment]
Does it happen in other repos?
Is `next-version` defined in the output of `gitversion /showConfig`?

The `major` version value seems to be derived from a date (8-16-2023).

**Comment by dnyandeobharambe** [comment]
> Does it happen in other repos? Is `next-version` defined in the output of `gitversion /showConfig`?
> 
> The `major` version value seems to be derived from a date (8-16-2023).

Some Other repos there is no issue to stuck for same version . But in all repos I can not bump patch, even if I provide correct command it go to update minor. But some repos have this issue too.

How this major version can derived from date? Name of branch is name: $(SourceBranchName)$(Date:.yyyyMMdd)$(Rev:.r)
does this cause issue? Other thing I am testing in feature branch

**Comment by BinToss** [comment]
>  $(SourceBranchName)$(Date:.yyyyMMdd)$(Rev:.r)
does this cause issue?

No, that wouldn't cause this issue.
I can only assume there's some Git data (i.e. tags, misc info) that is affecting the major version's variable. I also considered environment variables, but those wouldn't carry over to the CI environment...unless it's being exported earlier in the CI.

**Comment by dnyandeobharambe** [comment]
> > (SourceBranchName)(Date:.yyyyMMdd)$(Rev:.r)
> > does this cause issue?
> 
> No, that wouldn't cause this issue. I can only assume there's some Git data (i.e. tags, misc info) that is affecting the major version's variable. I also considered environment variables, but those wouldn't carry over to the CI environment...unless it's being exported earlier in the CI.

I am also thinking it could be git, but I do not have solution. Do you have any way to to clean it. I tried delete tag loacal and remote couple of times. It did not help me. Do you think this issue will continue after merge to main branch too? Any clue to resolve this.

**Comment by dnyandeobharambe** [comment]
> > > (SourceBranchName)(Date:.yyyyMMdd)$(Rev:.r)
> > > does this cause issue?
> > 
> > 
> > No, that wouldn't cause this issue. I can only assume there's some Git data (i.e. tags, misc info) that is affecting the major version's variable. I also considered environment variables, but those wouldn't carry over to the CI environment...unless it's being exported earlier in the CI.
> 
> I am also thinking it could be git, as I see it in other repo too even I said next-version: 1.0.0.It take date as major value.But I do not have solution. Do you have any way to to clean it. I tried delete tag local and remote couple of times. It did not help me. Do you think this issue will continue after merge to main branch too? Any clue to resolve this.
Note _ I am in feature branch and one repo I was able to get non-date. It is same way it implemented.
Do you think I should try to delete from azure devops and re-add it, is that could be issue? Is mode Mainline is not valid in this version. I do not see it , saying invalid with red mark . I try to change mode but no change.
Is this really date or just big number and is any way to reset it?  Somehow I was not able to increase it by major command ,but inbetween I get an error for currentversion step and after that it took next number. If this is number can I reset it and make it custom like start from 1.0.0

**Comment by BinToss** [comment]
If I had this issue, I would check for Git references via SmartGit (with a Personal license). If it doesn't show any obvious issues, I would start Googling Git's plumbing commands to see if Git has any commands to  help look for a tag or note.

If neither of those help diagnose the issue...
- Update GitVersion. Maybe the issue is fixed in 5.12 or 6.0's prereleases.
- Check if the erroneous value was caused by a GitVersion config value in a previous commit or tag.
- Check for the erroneous value in the environment variables.
- If the erroneous value is only in CI, check the CI's config's environment variables.

**Comment by dnyandeobharambe** [comment]
I explicitly not storing it in environment variables. How I can update Gitversion in azure devops pipeline?
Here I am sharing my config as well as code to get version.  Let me know if you find some thing fishy?
next-version: 1.0.0
major-version-bump-message: '\+semver:\s?(breaking|major)'
minor-version-bump-message: '\+semver:\s?(feature|minor)'
patch-version-bump-message: '\+semver:\s?(fix|patch)'
no-bump-message: '\+semver:\s?(none|skip)'
branches:
  release:
    regex: ^releases?[/-]
    tag: ''
  feature:
    regex: ^feature?[/-]
    tag: ''


 - task: gitversion/execute@0
      name: CalculateSemVer
      displayName: 'Calculate SemVer'
      inputs:
        useConfigFile: true
        configFilePath: './azure-pipelines/GitVersion.yml'
    - bash: |
        currentVersion=$(GitVersion.SemVer)
        echo "currentversion is $currentVersion"
        echo "##vso[task.setvariable variable=semversion;isOutput=true;]$currentVersion"
      name: SetOutputVariable

Following code I am using to read commit message

check_semver_format() {
            # Function definition here
            semver_pattern='\+semver:(breaking|major|feature|minor|fix|patch)'
            if [[ $1 =~ ^$semver_pattern ]]; then
                echo "startsWith"
            else
                if [[ $1 =~ $semver_pattern ]]; then
                    echo "contains"
                else
                    echo "doesNotMatch"
              fi
          fi
          }
          # Example usage
          given_string=$(git log --format=%B -n 1)
          result=$(check_semver_format "$given_string")
          echo "Result: $result"
          # Set a variable for use in subsequent jobs
          echo "##vso[task.setvariable variable=SemverResult;isOutput=true]$result"
        name: RunBashScript 


Note- As I am using Azure devops pipeline , I think my agent is new for every run and so it can not maintain any state in this variables?  You can correct me if I am wrong

I saw something this in log, is this tell some story
NextVersion in GitVersion configuration file: 1.0.0 with commit count source External Source
    INFO [03/28/24 22:42:04:20] Fallback base version: 0.1.0 with commit count source f74fcc191954fcbe0e4fb581e7ccb9180932b889
    INFO [03/28/24 22:42:04:32] Git tag '3082027.0.0': 3082027.0.0 with commit count source 2822b01e984b65296aa8ff36cbe81e836748949a
    INFO [03/28/24 22:42:04:32] Git tag '3082026.0.0': 3082026.0.0 with commit count source 3ada2dfd8cf6673644979fabbde05391d5cd3aea
    INFO [03/28/24 22:42:04:32] Git tag '3082025.2.0': 3082025.2.0 with commit count source 50bb9b516f92e6967f471248b289e76b2cc205e5
    INFO [03/28/24 22:42:04:32] Git tag '3082025.1.0': 3082025.1.0 with commit count source 003a163ac018b3c6cc7050f6feefb9c1b9998b95
    INFO [03/28/24 22:42:04:32] Git tag '3082025.0.0': 3082025.0.0 with commit count source bef360d130251fdf46203c31b9ec0d92ef1b0b80
    INFO [03/28/24 22:42:04:32] Git tag '3082024.0.0': 3082024.0.0 with commit count source fd73a6a755168499312bfc49d259c84279f1a286

**Comment by dnyandeobharambe** [comment]
here is complete log
[log-gitversion.txt](https://github.com/GitTools/GitVersion/files/14841469/log-gitversion.txt)

**Comment by dnyandeobharambe** [comment, marked as answer]
```yaml
mode: Mainline
major-version-bump-message: '\+semver:\s?(breaking|major)'
minor-version-bump-message: '\+semver:\s?(feature|minor)'
patch-version-bump-message: '\+semver:\s?(fix|patch)'
no-bump-message: '\+semver:\s?(none|skip)'
commit-message-incrementing: Enabled
ignore:
  commits-before: 2024-03-09T06:02:00
branches:
  main:
    regex: ^vertest$  # Matches your developement branch
    tag: ''  # No tag associated with the feature branch
    source-branches: []
```
